WORKSPACE

```diff
@@ -67,3 +67,21 @@ load("@rules_python_gazelle_plugin//:deps.bzl", _py_gazelle_deps = "gazelle_deps
 # This rule loads and compiles various go dependencies that running gazelle
 # for python requirements.
 _py_gazelle_deps()
+
+load("@python//3.11.1:defs.bzl", "interpreter")
+
+#####################
+# Install twine for our own runfiles wheel publishing
+# Eventually we might want to install twine automatically for users too, see:
+# See https://github.com/bazelbuild/rules_python/issues/1016
+load("@rules_python//python:pip.bzl", "pip_parse")
+
+pip_parse(
```

Comment:
Why are we unpacking twine differently to the rest of our tools in the repo that use external libraries? Feels inconsistent. I think all of the rules internal tools should load dependencies in the same private dependency closure.

Reply:
We don't use any tools with pip_parse yet - we exploded their dependency graphs into our `python/pip_install/dependencies.bzl` file.

Reply:
Precisely, why don't we do that here too?

Reply:
It has a really long list of transitives, see the `requirements.txt` introduced in this PR. I don't think we want to splat that into our repo for our own usage any more than a user would want to do that.

Reply:
The length of the transitives isn't really an issue tbh. It's easy to automate or generate that. Being aware of what they are also yields strong benefits so we ensure they are compatible on the same PYTHONPATH and also are pure python so we can use them more easily.

I feel that we should handle the "rule-private" dependencies of our tools consistently.

Reply:
I could imagine a helper like `py_twine_repositories` in repositories.bzl that does the 'pip_parse`, making this easier for users. I split that out to #1016 for now.